[addons][pvr] cleanups to match more coming C++ interface style and fix duplicated queue and VFS types

AlwinEsch opened the pull request:
## Description

This include some C++ rework changes, where also be can in independent. Again thought to reduce the others size and to have only changes related for C -> C++ parts there.

Commit 1:
------------------------------------------------
**Fix duplicated queue and VFS types**

Fixes Debian packaging's header-check failure

Commit 2:
------------------------------------------------
**[addons][pvr] separate menu hook calls and remove union use**

This is thought to make the calls cleaner, with the union before
was it always a bit ugly and unsafe.

Now have every type a own function where give his related data.
Further is on EPG menu hook the whole entry given.

Commit 3:
------------------------------------------------
**[addons][pvr] rename PVR_PROPERTIES to AddonProperties_PVR**

Done to match like on other addon types and to match his function tables.
Also is inside AddonInstance_PVR the "struct" before added to have "C"
conform.

Commit 4:
------------------------------------------------
**[addons][pvr] sort API function tables**

This is to have a better overview and with videoplayer parts on end.
If videoplayer inside is kicked out in future and done my inputstream
instance does it not break API.

Commit 5:
------------------------------------------------
**[addons][pvr] rename GetAddonCapabilities to GetCapabilities**

This is done to have his name more equal to the addon types and
prepared for coming C++ interface where included inside class.

Commit 6:
------------------------------------------------
**[addons][pvr] pass channel unique id to SignalStatus and GetDescrambleInfo**

This thought that addon identify his related channel, for the case it becomes
more as one time in future used. Further can it be also if only one identify
a related inputstream class where do his work.

Commit 7:
------------------------------------------------
**[addons][pvr] increase PVR API to 6.5.0**

## Motivation and Context
Come to end where the PVR addons soon on new interface and the system becomes cleaned up,
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

basilgello committed:
Fix duplicated queue and VFS types

Fixes Debian packaging's header-check failure:

rm -f debian/kodi-addons-dev/usr/include/kodi/xbmcclient.h
g++ -c debian/headers-check.c -DBUILD_KODI_ADDON \
    -Wdate-time -D_FORTIFY_SOURCE=2 -D_XBMC -g \
    -fdebug-prefix-map=/buildroot/sources/kodi/debian-source-package=. \
    -fstack-protector-strong -Wformat -Werror=format-security -O3 \
    -Idebian/kodi-addons-dev/usr/include \
    -o /dev/null
In file included
    from debian/kodi-addons-dev/usr/include/kodi/libKODI_guilib.h:16,
    from debian/headers-check.c:40:
debian/kodi-addons-dev/usr/include/kodi/libXBMC_addon.h:81:3:
    error: 'QUEUE_INFO' conflicts with a previous declaration
   QUEUE_INFO,
   ^~~~~~~~~~
In file included from debian/headers-check.c:13:
debian/kodi-addons-dev/usr/include/kodi/General.h:73:3:
   note: previous declaration 'QueueMsg QUEUE_INFO'
   QUEUE_INFO,
   ^~~~~~~~~~
In file included
    from debian/kodi-addons-dev/usr/include/kodi/libKODI_guilib.h:16,
    from debian/headers-check.c:40:
debian/kodi-addons-dev/usr/include/kodi/libXBMC_addon.h:82:3:
    error: 'QUEUE_WARNING' conflicts with a previous declaration
   QUEUE_WARNING,
   ^~~~~~~~~~~~~
In file included from debian/headers-check.c:13:
debian/kodi-addons-dev/usr/include/kodi/General.h:75:3:
    note: previous declaration 'QueueMsg QUEUE_WARNING'
   QUEUE_WARNING,
   ^~~~~~~~~~~~~
In file included
    from debian/kodi-addons-dev/usr/include/kodi/libKODI_guilib.h:16,
    from debian/headers-check.c:40:
debian/kodi-addons-dev/usr/include/kodi/libXBMC_addon.h:83:3:
    error: 'QUEUE_ERROR' conflicts with a previous declaration
   QUEUE_ERROR
   ^~~~~~~~~~~

etc ...

Signed-off-by: Vasyl Gello <vasek.gello@gmail.com>

AlwinEsch committed:
[addons][pvr] increase PVR API to 6.5.0

AlwinEsch committed:
[addons][pvr] rename PVR_PROPERTIES to AddonProperties_PVR

Done to match like on other addon types and to match his function tables.
Also is inside AddonInstance_PVR the "struct" before added to have "C"
conform.

AlwinEsch committed:
[addons][pvr] rename SignalStatus to GetSignalStatus

This thought to have his name equal with "Get" to other functions in this group.
Specially with the same place defined GetDescrampleInfo.

phunkyfish commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h:
```diff
@@ -224,7 +201,7 @@ namespace ADDON
      * @param format The format of the message to pass to display in XBMC.
      */
-    void QueueNotification(const queue_msg_t type, const char *format, ... )
+    void QueueNotification(const QueueMsg type, const char *format, ... )
```
Can or should this be a reference?

phunkyfish replied:
Also, `const char* format, ...`

Left align `*`

AlwinEsch replied:
Ah thanks, has not thought about :smile:

phunkyfish commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h:
```diff
@@ -502,21 +502,23 @@ extern "C"
    *          Return PVR_ERROR_NOT_IMPLEMENTED if this add-on won't provide this function.
    */
-  PVR_ERROR SignalStatus(PVR_SIGNAL_STATUS& signalStatus);
+  PVR_ERROR SignalStatus(int channelUid, struct PVR_SIGNAL_STATUS* signalStatus);
```
is the struct keyword required? I don't see it elsewhere in the API?

phunkyfish commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h:
```diff
@@ -502,21 +502,23 @@ extern "C"
    *          Return PVR_ERROR_NOT_IMPLEMENTED if this add-on won't provide this function.
    */
-  PVR_ERROR GetDescrambleInfo(PVR_DESCRAMBLE_INFO* descrambleInfo);
+  PVR_ERROR GetDescrambleInfo(int channelUid, struct PVR_DESCRAMBLE_INFO* descrambleInfo);
```
is the struct keyword required? I don't see it elsewhere in the API?

AlwinEsch replied:
Yes and no, also forgot add on `PVR_ERROR (__cdecl* GetDescrambleInfo)(int, PVR_DESCRAMBLE_INFO*);`

As the addons use currently only C++ is it not required, but to have it as "C" only becomes it needed. But I remove from there and becomes done on rework request.

This request for here is mostly thought to reduce amount of changed places outside of PVR addon header.

phunkyfish commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h:
```diff
@@ -30,7 +30,7 @@ extern "C"
    * @remarks Valid implementation required.
    */
-  PVR_ERROR GetAddonCapabilities(PVR_ADDON_CAPABILITIES *pCapabilities);
+  PVR_ERROR GetCapabilities(PVR_ADDON_CAPABILITIES *pCapabilities);
```
Left alight `*`

phunkyfish commented:
@basilgello yours is the first commit in this PR.

basilgello commented:
I noticed that @phunkyfish :) But thanks for the message!

ksooo commented on xbmc/pvr/addons/PVRClient.cpp:
```diff
@@ -1379,79 +1379,71 @@ std::shared_ptr<CPVRClientMenuHooks> CPVRClient::GetMenuHooks()
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = PVR_MENUHOOK_CHANNEL;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallChannelMenuHook(&menuHook, &tag);
```
Why do I have to set `menuHook.category`? The called method already defines the semantics, so no need to say which "category". It's clear that it must be "channel" when calling `CallChannelMenuHook`. Same comment applies to the other "Call*MenuHook" calls.

ksooo commented on xbmc/pvr/addons/PVRClient.cpp:
```diff
@@ -1379,79 +1379,71 @@ std::shared_ptr<CPVRClientMenuHooks> CPVRClient::GetMenuHooks()
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = bDeleted ? PVR_MENUHOOK_DELETED_RECORDING : PVR_MENUHOOK_RECORDING;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallRecordingMenuHook(&menuHook, &tag);
```
Looks like here comes `menuHook.category` (see my previous comment) into play, but why this inconsistency? Either we introduce `CallRecordingMenuHook` and `CallDeletedRecordingMenuHook` and get rid of `PVR_MENUHOOK::category` or ... as I don't get the benefit of this whole change to make 5 (6?) `Call*MenuHook` functions out of the one API call we had before ... just revert this whole change?

AlwinEsch replied:
Have this related commit removed and becomes done on C++ rework.

ksooo reviewed (commented):
To be honest; I don't understand the ratio behind the menu hook related changes. They are not consistent (wrt. PVR_MENUHOOK::category usage), introduce 5 (6?) API functions where formerly one was sufficient - without any functional gain.

AlwinEsch committed:
[addons][pvr] pass channel unique id to SignalStatus and GetDescrambleInfo

This thought that addon identify his related channel, for the case it becomes
more as one time in future used. Further can it be also if only one identify
a related inputstream class where do his work.

AlwinEsch committed:
[addons][pvr] sort API function tables

This is to have a better overview and with videoplayer parts on end.
If videoplayer inside is kicked out in future and done my inputstream
instance does it not break API.

AlwinEsch committed:
[addons][pvr] rename GetAddonCapabilities to GetCapabilities

This is done to have his name more equal to the addon types and
prepared for coming C++ interface where included inside class.

AlwinEsch commented:
All addons are prepared for this request, OK to bring in (if jenkins have a happy day today)?

jenkins build this please

phunkyfish reviewed (approved):
Yes, fine by me.

phunkyfish commented:
jenkins build this please

basilgello commented:
@AlwinEsch https://github.com/kodi-pvr/pvr.argustv/pull/103 still open.
